test/other/test_pthread_mutex_deadlock.c

```diff
@@ -0,0 +1,20 @@
+#include <assert.h>
+#include <pthread.h>
+#include <stdbool.h>
+#include <stdio.h>
+
+pthread_mutex_t m = PTHREAD_MUTEX_INITIALIZER;
+
+int main() {
+  printf("in main\n");
```

Comment:
```suggestion
  puts("in main");
```

Reply:
We normally don't shy away from using printf for this kind of thing in test code.